contracts/GovNftSugar.vy

```diff
@@ -52,121 +66,154 @@ interface IGovNFT:
   def locks(_govnft_id: uint256) -> Lock: view
   def unclaimed(_govnft_id: uint256) -> uint256: view
   def locked(_govnft_id: uint256) -> uint256: view
+  def name() -> String[100]: view
+  def symbol() -> String[100]: view
+  def owner() -> address: view
 
 interface IToken:
   def delegates(_account: address) -> address: view
 
 # Vars
-nft: public(IGovNFT)
+factory: public(IGovNFTFactory) # GovNFT factory
+implementations: public(DynArray[address, MAX_COLLECTIONS]) # implementations (collections) created by factory
 
 # Methods
 
 @external
-def __init__(_nft: address):
+def __init__(_factory: address):
   """
-  @dev Set up the GovNFT contract
+  @dev Set up the GovNFTFactory contract
   """
-  self.nft = IGovNFT(_nft)
+  self.factory = IGovNFTFactory(_factory)
+  self.implementations = self.factory.govNFTs()
```

Comment:
Btw, if we're caching this, does it mean that we'll have to redeploy the sugar when there are new collections?

What's the goal here?

Reply:
ty ty, looks good to me